### ISSUE TITLE
xmi2shex doesn't work

### ISSUE THREAD

**Issue by milkbread** [weso/rdfshape-client#114]
Hi,

the [UML conversion to ShEx](https://rdfshape.weso.es/xmi2shex) is currently broken. Even when I try to run the [example](https://rdfshape.weso.es/xmi2shex?schemaTargetEngine=Shumlex&schemaTargetFormat=XMI&uml=%0A%3C%3Fxml%20version%3D%221.0%22%20encoding%3D%22UTF-8%22%3F%3E%0A%3Cuml%3AModel%20xmi%3Aversion%3D%222.1%22%20xmlns%3Axmi%3D%22http%3A%2F%2Fschema.omg.org%2Fspec%2FXMI%2F2.1%22%20xmlns%3Auml%3D%22http%3A%2F%2Fwww.eclipse.org%2Fuml2%2F3.0.0%2FUML%22%0A%20xmi%3Aid%3D%22kzimxktg%22%20name%3D%22ShExGeneratedXMI%22%3E%0A%3CpackagedElement%20xmi%3Atype%3D%22uml%3AClass%22%20xmi%3Aid%3D%22kzimxktm%22%20name%3D%22%3AUser%22%3E%0A%09%3CownedAttribute%20xmi%3Aid%3D%22kzimxktn%22%20name%3D%22schema%3Aname%22%20visibility%3D%22public%22%20isUnique%3D%22false%22%3E%0A%09%09%3Ctype%20xmi%3Atype%3D%22uml%3APrimitiveType%22%20href%3D%22pathmap%3A%2F%2FUML_LIBRARIES%2FUMLPrimitiveTypes.library.uml%23String%22%3E%0A%09%09%3C%2Ftype%3E%0A%09%3C%2FownedAttribute%3E%0A%09%3CownedAttribute%20xmi%3Aid%3D%22kzimxktp%22%20name%3D%22schema%3AbirthDate%22%20visibility%3D%22public%22%20isUnique%3D%22false%22%3E%0A%09%09%3Ctype%20xmi%3Atype%3D%22uml%3APrimitiveType%22%20href%3D%22pathmap%3A%2F%2FUML_LIBRARIES%2FUMLPrimitiveTypes.library.uml%23Date%22%3E%0A%09%09%3C%2Ftype%3E%0A%09%09%3ClowerValue%20xmi%3Atype%3D%22uml%3ALiteralInteger%22%20xmi%3Aid%3D%22kzimxkto%22%20%2F%3E%0A%09%3C%2FownedAttribute%3E%0A%09%3CownedAttribute%20xmi%3Atype%3D%22uml%3AProperty%22%20xmi%3Aid%3D%22kzimxktq%22%20name%3D%22schema%3Agender%22%20visibility%3D%22public%22%20type%3D%22kzimxktr%22%20isUnique%3D%22true%22%3E%0A%09%3C%2FownedAttribute%3E%0A%09%3CownedAttribute%20xmi%3Aid%3D%22kzimxktq%22%20name%3D%22schema%3Agender%22%20visibility%3D%22public%22%20isUnique%3D%22false%22%3E%0A%09%09%3Ctype%20xmi%3Atype%3D%22uml%3APrimitiveType%22%20href%3D%22pathmap%3A%2F%2FUML_LIBRARIES%2FUMLPrimitiveTypes.library.uml%23String%22%3E%0A%09%09%3C%2Ftype%3E%0A%09%3C%2FownedAttribute%3E%0A%09%3CownedAttribute%20xmi%3Atype%3D%22uml%3AProperty%22%20xmi%3Aid%3D%22kzimxktu%22%20name%3D%22schema%3Aknows%22%20visibility%3D%22public%22%20type%3D%22kzimxktv%22%20isUnique%3D%22true%22%3E%0A%09%3C%2FownedAttribute%3E%0A%09%3CownedAttribute%20xmi%3Aid%3D%22kzimxktx%22%20name%3D%22schema%3Aknows%22%20visibility%3D%22public%22%20type%3D%22kzimxktm%22%20association%3D%22kzimxkty%22%20%3E%3C%2FownedAttribute%3E%0A%3C%2FpackagedElement%3E%0A%3CpackagedElement%20xmi%3Atype%3D%22uml%3AAssociation%22%20xmi%3Aid%3D%22kzimxkty%22%20memberEnd%3D%22kzimxktx%20kzimxktz%22%3E%0A%09%3CownedEnd%20xmi%3Aid%3D%22kzimxktz%22%20visibility%3D%22public%22%20type%3D%22kzimxktm%22%20association%3D%22kzimxkty%22%2F%3E%0A%3C%2FpackagedElement%3E%0A%3CpackagedElement%20xmi%3Atype%3D%22uml%3AEnumeration%22%20xmi%3Aid%3D%22kzimxktl%22%20name%3D%22Prefixes%22%3E%0A%09%3CownedLiteral%20xmi%3Aid%3D%22kzimxkth%22%20name%3D%22prefix%20%3A%20%26lt%3Bhttp%3A%2F%2Fexample.org%2F%3E%22%2F%3E%0A%09%3CownedLiteral%20xmi%3Aid%3D%22kzimxkti%22%20name%3D%22prefix%20schema%3A%20%26lt%3Bhttp%3A%2F%2Fschema.org%2F%3E%22%2F%3E%0A%09%3CownedLiteral%20xmi%3Aid%3D%22kzimxktj%22%20name%3D%22prefix%20xsd%3A%20%26lt%3Bhttp%3A%2F%2Fwww.w3.org%2F2001%2FXMLSchema%23%3E%22%2F%3E%0A%09%3CownedLiteral%20xmi%3Aid%3D%22kzimxktk%22%20name%3D%22base%20%26lt%3Bhttp%3A%2F%2Fexample.org%2F%3E%22%2F%3E%0A%3C%2FpackagedElement%3E%0A%3CpackagedElement%20xmi%3Atype%3D%22uml%3AEnumeration%22%20xmi%3Aid%3D%22kzimxktr%22%20name%3D%22schema%3Agender%22%3E%0A%09%3CownedLiteral%20xmi%3Aid%3D%22kzimxku0%22%20name%3D%22schema%3AMale%22%2F%3E%0A%09%3CownedLiteral%20xmi%3Aid%3D%22kzimxku1%22%20name%3D%22schema%3AFemale%22%2F%3E%0A%3C%2FpackagedElement%3E%0A%3CpackagedElement%20xmi%3Atype%3D%22uml%3APrimitiveType%22%20xmi%3Aid%3D%22kzimxktv%22%20name%3D%22IRI%22%2F%3E%0A%3C%2Fuml%3AModel%3E%0A%20%20&umlFormat=XMI&umlSource=byText)...I only get an error response:

```
Error response from https://api.rdfshape.weso.es
An error has occurred while creating the ShEx equivalent. Check your inputs.
```

Cheers
Ralf

**Comment by labra** [comment]
You are right. There is indeed some functionality that has been lost in the recent changes of rdfshape. 

I keep the issue open until we have time or resources to update rdfshape and repair that functionality, although at this moment, it may take some time. 

The xmi2shex functionality was implemented using this library: https://github.com/weso/shumlex
